integration/commands_administration_test.go

```diff
@@ -764,6 +764,111 @@ func TestCommandsAdministrationDBStatsEmptyWithScale(t *testing.T) {
 	// https://github.com/FerretDB/FerretDB/issues/727
 }
 
+func TestCommandsAdministrationRenameCollection(t *testing.T) {
+	t.Parallel()
+	insertCollections := []string{"foo", "none"}
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		collection       string
+		targetCollection string
+		expected         bson.D
+		err              *mongo.CommandError
+	}{
+		"Rename": {
+			collection:       "foo",
+			targetCollection: "bar",
+			expected:         bson.D{{"ok", float64(1)}},
+		},
+		"RenameSame": {
+			collection:       "foo",
+			targetCollection: "foo",
+			err: &mongo.CommandError{
+				Code:    20,
+				Name:    "IllegalOperation",
+				Message: `Can't rename a collection to itself`,
+			},
+		},
+		"RenameDuplicate": {
+			collection:       "foo",
+			targetCollection: "none",
+			err: &mongo.CommandError{
+				Code:    48,
+				Name:    "NamespaceExists",
+				Message: `target namespace exists`,
+			},
+		},
+		"RenameVerifyMetadata": {
+			collection:       "foo",
+			targetCollection: "bar",
+			expected:         bson.D{{"ok", float64(1)}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx, collection := setup.Setup(t, shareddata.Bools)
+
+			db := collection.Database()
+			require.NoError(t, db.Drop(ctx))
+
+			require.NotEmpty(t, insertCollections)
+			for _, coll := range insertCollections {
+				require.NoError(t, db.CreateCollection(ctx, coll))
+			}
+
+			var actual bson.D
+
+			if name == "RenameVerifyMetadata" {
```

Comment:
Instead of using test name, which can easily change we should add a test parameter. By the way, why do we have to insert instead of using initial `shareddata.Bools`?

Example
```
for name, tc := range map[string]struct { //nolint:vet // for readability
		collection       string
		targetCollection string
		expected         bson.D
		err              *mongo.CommandError
toInsert []bson.D
	}
```

then 
```
if toInsert != nil{
// insert things
```

Reply:
That's so elegant, thank you. Will utilize that approach.